app/docker/models/template.js

```diff
@@ -40,11 +40,19 @@ function TemplateViewModel(data) {
   this.Ports = [];
   if (data.ports) {
     this.Ports = data.ports.map(function (p) {
-      var portAndProtocol = _.split(p, '/');
-      return {
-        containerPort: portAndProtocol[0],
-        protocol: portAndProtocol[1]
-      };
+      var portConf = {};
+      
+      if(_.isObject(p)){
```

Comment:
Instead of an if/else, you can return an object directly in the if statement block:

```javascript
if (_.isObject(p)) {
  return {
    containerPort: p.container,
    protocol: p.protocol || 'tcp',
    hostPort: p.host || null,
  };
}
```